lib/request-state.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var Map = require('immutable').Map;
+var History = require('./history');
+
+/**
+ * A Request State is an immutable state corresponding to a request processing or processed by the system.
+ *
+ * @param {string} status - The current status of the request
+ * @param {Map.<string, Job>} jobs - A map of all the jobs related to this request
+ * @param {History} history - A human-readable history for this request
+ * @param {Map.<string, *>} metadata - The metadata extracted by Specberus for this request
+ */
+function RequestState(status, jobs, history, metadata) {
+  if (typeof this !== 'object') {
+    throw new TypeError('RequestState must be constructed via new');
+  }
+
+  this.status = typeof status !== 'undefined' ? status : '';
```

Comment:
`typeof x!== 'undefined'` is one of these things that are generally recommended but that I have to look up to remember why.

In that case, this is a bit of a stretch, but `undefined` can be overwritten (apparently, this is not true since ES5 though) while `typeof` is a built-in operator that always returns a `string`.

[This page](http://stackoverflow.com/questions/3390396/how-to-check-for-undefined-in-javascript) has some interesting stuff. I will not argue over what we should use: I am simply not experienced enough to take a stand. Please let me know what you prefer and I'll align with it.